arrow-libs/core/arrow-autoclose/src/commonTest/kotlin/arrow/AutoCloseTest.kt

```diff
@@ -136,6 +135,23 @@ class AutoCloseTest {
     res.isActive() shouldBe false
   }
 
+  @Test
+  fun closeTheAutoScopeOnNonLocalReturn() = runTest {
+    val wasActive = CompletableDeferred<Boolean>()
+    val res = Resource()
+
+    run {
+      autoCloseScope {
+        val r = install(res)
+        wasActive.complete(r.isActive())
+        return@run
+      }
+    }
+
+    wasActive.await() shouldBe true
```

Comment:
if the test fails, `.await()` will suspend until the 1minute timeout. It would be nice to have a `wasActive shouldHaveCompletedWith true` matcher that checks `Job.isComplete` before calling `.await()` (or `.getCompleted()`)

this also applies to a number of tests throughout (noticed when I tried breaking things)

Reply:
I was copying the style of the other tests in the file, but yes it might be worth it to make those tests fail-fast. I'll make the necessary changes (likely just using a `lateinit var`)

Reply:
Done!